### ISSUE TITLE
Popuplyrics cannot find lyrics anymore

### ISSUE THREAD

**Issue by frallatta** [Patrick-orz/PopupLyrics-Plus#9]
**Are you using the LATEST version?**
Yes

**Additional context**
Hello, i think current popuplyricsplus didn't compatible with current spotify version,
popuplyricsplus cannot find any lyrics after spotify updates

**Comment by Patrick-orz** [comment]
Using spicetify v2.37 and the latest version of spotify, all popuplyricsplus's functions work fine on my end (except for the stupidly slow launch speed for english translations...).

![image](https://github.com/user-attachments/assets/8cd01cdc-f888-48c3-8dd6-ef3e10279863)
![image](https://github.com/user-attachments/assets/480a5e18-0428-451d-a846-1297cf8bb5be)
![image](https://github.com/user-attachments/assets/bb4ab8e1-1fe6-418c-bbb8-c2fe79abefde)

I will close the issue for now, and if the issue persists for you, please provide detailed steps of reproduction and information on your environment. BTW, I wonder if the official popuplyrics extension works for you?

**Comment by Patrick-orz** [comment]
UPDATE: I realized I wasn't using the latest version of spotify when making the previous post. After reinstalling spotify and updating it to the newest version, I believe spicetify automatically downgraded spotify back into an older version when I did _"spicetify backup apply"_ and such. **Anyhow, everything worked fine on my presumably automatically downgraded version of the latest spotify.** Note that I didn't manually or intentionally downgrade my client, all I did was inject spicetify as usual. I couldn't have used popuplyricsplus or anything related to spicetify without doing so.

I reckon spicetify downgraded my spotify because it is now called "spotify (Old)" rather than just "spotify". I remember reading about some incompatibility between spicetify and latest versions of spotify. I can't recall clearly as it has been some time. I assume spicetify should downgrade incompatible spotify versions automatically (as it potentially did for me).

In general, there is a lot of uncertainty on this topic, but nevertheless the issue isn't reproducible on my end. Therefore, this issue would still stay closed until more information is presented on your end. I am happy to help!

_Potential Questions Answer:_
_- Does it seem like spicetify downgraded your spotify?_
_- Does original popuplyrics work?_

**Comment by frallatta** [comment]
Im using this version : 
  
```
Spotify for Windows (64 bit)
1.2.43.420.gc7b23f70
Spicetify v2.37.0
``` 

already tried with current popuplyricsplus but stuck on finding lyrics. and when tried with official popuplyrics it works well.
i also try to look into the codes and find the source of the lyrics already change.
maybe it was the root of the problem ?

**Comment by Patrick-orz** [comment]
I think you are right that the source of lyrics could be the issue, check out this [issue](https://github.com/spicetify/cli/issues/3056) targeted towards the official PopupLyrics extension. The issue is later resolved by this [commit](https://github.com/spicetify/cli/commit/41474fa0195878ff892e27c8a5b0b9de3376e90a).

The same issue might be occurring here? See if you can get lyrics from any other services such as Netease(CN) or Musixmatch, since the issue above seems to only be with the Spotify lyrics service.

I don't have a great amount of time at hand right now, but I would look into reproducing and patching this issue once available (very soon) :P

In the meantime, I encourage you to try implementing the fix committed to the official popuplyrics extension. If that resolves this issue, submit a PR! That would be greatly appreciated :)

**Comment by Patrick-orz** [comment]
Should be resolved with the merge of #10.
